lib/vssh/libssh2.c

```diff
@@ -2003,6 +2003,7 @@ static CURLcode ssh_statemachine(struct Curl_easy *data, bool *block)
       if(rc > 0) {
         /* It seems that this string is not always NULL terminated */
         sshp->readdir_filename[rc] = '\0';
+        Curl_safefree(sshc->homedir);
```

Comment:
As it gets assigned immediately afterwards, you might as well use a plain free:

```suggestion
        free(sshc->homedir);
```